core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -311,26 +312,35 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
     if(_entries == 0)
       return (-1, -1)
 
+    def binarySearch(begin: Int, end: Int) : (Int, Int) = {
+      // binary search for the entry
+      var lo = begin
+      var hi = end
+      while(lo < hi) {
+        val mid = ceil(hi/2.0 + lo/2.0).toInt
```

Comment:
This line is copied from existing code. i didn't change it.

Reply:
It should be more efficient to use ints than floating point here, right?

Reply:
This is an existing line. I did touch it. If you want to improve it, you should create another pull request and move the discussion to the new pull request. Fixing this line has nothing to do with this ticket.

Reply:
OK.  I filed https://github.com/apache/kafka/pull/5378 to fix this.

Reply:
@ying-zheng because this is a trivial change and you are touching this line, I think we should change it to use the best implementation in this PR. More PRs is obviously more overhead for everyone involved. :-)

Reply:
I don't know if the change is that trivial.  We probably need to think a bit about overflow, rounding, etc.  It's fine for Ying Z. to push back a bit and ask us to do it in a separate PR.